File: NegocioCSharp/RecuperarClave.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
namespace LiquidacionSueldos
{
    namespace Negocio
    {
        public partial class RecuperarClave
        {
            private Datos.Gestor ocdGestor = new Datos.Gestor();
            private DataTable Fila;

            private DataTable Tabla = new DataTable();
            private int _rclId;
            public int rclId
            {
                get { return _rclId; }
                set { _rclId = value; }
            }

            private System.DateTime _rclFecha;
            public System.DateTime rclFecha
            {
                get { return _rclFecha; }
                set { _rclFecha = value; }
            }

            private string _rclUsuario;
            public string rclUsuario
            {
                get { return _rclUsuario; }
                set { _rclUsuario = value; }
            }

            private string _rclEmailRecuperacion;
            public string rclEmailRecuperacion
            {
                get { return _rclEmailRecuperacion; }
                set { _rclEmailRecuperacion = value; }
            }

            private string _rclUsuarioEncriptado;
            public string rclUsuarioEncriptado
            {
                get { return _rclUsuarioEncriptado; }
                set { _rclUsuarioEncriptado = value; }
            }

            private string _rclEmailRecuperacionEncriptado;
            public string rclEmailRecuperacionEncriptado
            {
                get { return _rclEmailRecuperacionEncriptado; }
                set { _rclEmailRecuperacionEncriptado = value; }
            }

            private bool _rclActivo;
            public bool rclActivo
            {
                get { return _rclActivo; }
                set { _rclActivo = value; }
            }

            private System.DateTime _rclFechaHoraCreacion;
            public System.DateTime rclFechaHoraCreacion
            {
                get { return _rclFechaHoraCreacion; }
                set { _rclFechaHoraCreacion = value; }
            }

            private System.DateTime _rclFechaHoraUltimaModificacion;
            public System.DateTime rclFechaHoraUltimaModificacion
            {
                get { return _rclFechaHoraUltimaModificacion; }
                set { _rclFechaHoraUltimaModificacion = value; }
            }

            private int _usuIdCreacion;
            public int usuIdCreacion
            {
                get { return _usuIdCreacion; }
                set { _usuIdCreacion = value; }
            }

            private int _usuIdUltimaModificacion;
            public int usuIdUltimaModificacion
            {
                get { return _usuIdUltimaModificacion; }
                set { _usuIdUltimaModificacion = value; }
            }

            public RecuperarClave()
            {
                try
                {
                    this.rclId = 0;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public RecuperarClave(int rclId)
            {
                try
                {
                    Fila = new DataTable();
                    Fila = ocdGestor.EjecutarReader("[RecuperarClave.ObtenerUno]", new object[,] { {
                        "@rclId",
                        rclId
                    } });

                    if (Fila.Rows.Count > 0)
                    {
                        if (Fila.Rows[0]["rclId"].ToString().Trim().Length > 0)
                        {
                            this.rclId = Convert.ToInt32(Fila.Rows[0]["rclId"]);
                        }
                        else
                        {
                            this.rclId = 0;
                        }

                        if (Fila.Rows[0]["rclFecha"].ToString().Trim().Length > 0)
                        {
                            this.rclFecha = Convert.ToDateTime(Fila.Rows[0]["rclFecha"]);
                        }
                        else
                        {
                            this.rclFecha = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["rclUsuario"].ToString().Trim().Length > 0)
                        {
                            this.rclUsuario = Convert.ToString(Fila.Rows[0]["rclUsuario"]);
                        }
                        else
                        {
                            this.rclUsuario = "";
                        }

                        if (Fila.Rows[0]["rclEmailRecuperacion"].ToString().Trim().Length > 0)
                        {
                            this.rclEmailRecuperacion = Convert.ToString(Fila.Rows[0]["rclEmailRecuperacion"]);
                        }
                        else
                        {
                            this.rclEmailRecuperacion = "";
                        }

                        if (Fila.Rows[0]["rclUsuarioEncriptado"].ToString().Trim().Length > 0)
                        {
                            this.rclUsuarioEncriptado = Convert.ToString(Fila.Rows[0]["rclUsuarioEncriptado"]);
                        }
                        else
                        {
                            this.rclUsuarioEncriptado = "";
                        }

                        if (Fila.Rows[0]["rclEmailRecuperacionEncriptado"].ToString().Trim().Length > 0)
                        {
                            this.rclEmailRecuperacionEncriptado = Convert.ToString(Fila.Rows[0]["rclEmailRecuperacionEncriptado"]);
                        }
                        else
                        {
                            this.rclEmailRecuperacionEncriptado = "";
                        }

                        if (Fila.Rows[0]["rclFechaHoraCreacion"].ToString().Trim().Length > 0)
                        {
                            this.rclFechaHoraCreacion = Convert.ToDateTime(Fila.Rows[0]["rclFechaHoraCreacion"]);
                        }
                        else
                        {
                            this.rclFechaHoraCreacion = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["rclFechaHoraUltimaModificacion"].ToString().Trim().Length > 0)
                        {
                            this.rclFechaHoraUltimaModificacion = Convert.ToDateTime(Fila.Rows[0]["rclFechaHoraUltimaModificacion"]);
                        }
                        else
                        {
                            this.rclFechaHoraUltimaModificacion = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["rclActivo"].ToString().Trim().Length > 0)
                        {
                            this.rclActivo = (Convert.ToInt32(Fila.Rows[0]["rclActivo"]) == 1 ? true : false);
                        }
                        else
                        {
                            this.rclActivo = false;
                        }

                        if (Fila.Rows[0]["usuIdCreacion"].ToString().Trim().Length > 0)
                        {
                            this.usuIdCreacion = Convert.ToInt32(Fila.Rows[0]["usuIdCreacion"]);
                        }
                        else
                        {
                            this.usuIdCreacion = 0;
                        }

                        if (Fila.Rows[0]["usuIdUltimaModificacion"].ToString().Trim().Length > 0)
                        {
                            this.usuIdUltimaModificacion = Convert.ToInt32(Fila.Rows[0]["usuIdUltimaModificacion"]);
                        }
                        else
                        {
                            this.usuIdUltimaModificacion = 0;
                        }

                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public RecuperarClave(int rclId, System.DateTime rclFecha, string rclUsuario, string rclEmailRecuperacion, string rclUsuarioEncriptado, string rclEmailRecuperacionEncriptado, bool rclActivo, System.DateTime rclFechaHoraCreacion, System.DateTime rclFechaHoraUltimaModificacion, int IusuIdCreacion,
            int IusuIdUltimaModificacion)
            {
                try
                {
                    this.rclId = rclId;
                    this.rclFecha = rclFecha;
                    this.rclUsuario = rclUsuario;
                    this.rclEmailRecuperacion = rclEmailRecuperacion;
                    this.rclUsuarioEncriptado = rclUsuarioEncriptado;
                    this.rclEmailRecuperacionEncriptado = rclEmailRecuperacionEncriptado;
                    this.rclActivo = rclActivo;
                    this.rclFechaHoraCreacion = rclFechaHoraCreacion;
                    this.rclFechaHoraUltimaModificacion = rclFechaHoraUltimaModificacion;
                    this.usuIdCreacion = usuIdCreacion;
                    this.usuIdUltimaModificacion = usuIdUltimaModificacion;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }


            public DataTable ObtenerConValoresEncriptados(string UsuarioEncriptado, string EmailEncriptado)
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[RecuperarClave.ObtenerConValoresEncriptados]", new object[,] {
                        {
                            "@UsuarioEncriptado",
                            UsuarioEncriptado
                        },
                        {
                            "@EmailEncriptado",
                            EmailEncriptado
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerTodo()
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[RecuperarClave.ObtenerTodo]", new object[,]{});
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerUno(int rclId)
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[RecuperarClave.ObtenerUno]", new object[,] { {
                        "@rclId",
                        rclId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public void Actualizar(int rclId, System.DateTime rclFecha, string rclUsuario, string rclEmailRecuperacion, string rclUsuarioEncriptado, string rclEmailRecuperacionEncriptado, bool rclActivo, int usuIdCreacion, int usuIdUltimaModificacion, System.DateTime rclFechaHoraCreacion,
            System.DateTime rclFechaHoraUltimaModificacion)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[RecuperarClave.Actualizar]", new object[,] {
                        {
                            "@rclId",
                            rclId
                        },
                        {
                            "@rclFecha",
                            rclFecha
                        },
                        {
                            "@rclUsuario",
                            rclUsuario
                        },
                        {
                            "@rclEmailRecuperacion",
                            rclEmailRecuperacion
                        },
                        {
                            "@rclUsuarioEncriptado",
                            rclUsuarioEncriptado
                        },
                        {
                            "@rclEmailRecuperacionEncriptado",
                            rclEmailRecuperacionEncriptado
                        },
                        {
                            "@rclActivo",
                            rclActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@rclFechaHoraCreacion",
                            rclFechaHoraCreacion
                        },
                        {
                            "@rclFechaHoraUltimaModificacion",
                            rclFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Eliminar(int rclId)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[RecuperarClave.Eliminar]", new object[,] { {
                        "@rclId",
                        rclId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Insertar(System.DateTime rclFecha, string rclUsuario, string rclEmailRecuperacion, string rclUsuarioEncriptado, string rclEmailRecuperacionEncriptado, bool rclActivo, int usuIdCreacion, int usuIdUltimaModificacion, System.DateTime rclFechaHoraCreacion, System.DateTime rclFechaHoraUltimaModificacion)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[RecuperarClave.Insertar]", new object[,] {
                        {
                            "@rclFecha",
                            rclFecha
                        },
                        {
                            "@rclUsuario",
                            rclUsuario
                        },
                        {
                            "@rclEmailRecuperacion",
                            rclEmailRecuperacion
                        },
                        {
                            "@rclUsuarioEncriptado",
                            rclUsuarioEncriptado
                        },
                        {
                            "@rclEmailRecuperacionEncriptado",
                            rclEmailRecuperacionEncriptado
                        },
                        {
                            "@rclActivo",
                            rclActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@rclFechaHoraCreacion",
                            rclFechaHoraCreacion
                        },
                        {
                            "@rclFechaHoraUltimaModificacion",
                            rclFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void InsertarCustom(string Usuario, string Email, string UsuarioEncriptado, string EmailEncriptado)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[RecuperarClave.InsertarCustom]", new object[,] {
                        {
                            "@Usuario",
                            Usuario
                        },
                        {
                            "@Email",
                            Email
                        },
                        {
                            "@UsuarioEncriptado",
                            UsuarioEncriptado
                        },
                        {
                            "@EmailEncriptado",
                            EmailEncriptado
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Actualizar()
            {
                try
                {
                    if (this.rclId != 0)
                    {
                        ocdGestor.EjecutarNonQuery("[RecuperarClave.Actualizar]", new object[,] {
                            {
                                "@rclId",
                                rclId
                            },
                            {
                                "@rclFecha",
                                rclFecha
                            },
                            {
                                "@rclUsuario",
                                rclUsuario
                            },
                            {
                                "@rclEmailRecuperacion",
                                rclEmailRecuperacion
                            },
                            {
                                "@rclUsuarioEncriptado",
                                rclUsuarioEncriptado
                            },
                            {
                                "@rclEmailRecuperacionEncriptado",
                                rclEmailRecuperacionEncriptado
                            },
                            {
                                "@rclActivo",
                                rclActivo
                            },
                            {
                                "@usuIdCreacion",
                                usuIdCreacion
                            },
                            {
                                "@usuIdUltimaModificacion",
                                usuIdUltimaModificacion
                            },
                            {
                                "@rclFechaHoraCreacion",
                                rclFechaHoraCreacion
                            },
                            {
                                "@rclFechaHoraUltimaModificacion",
                                rclFechaHoraUltimaModificacion
                            }
                        });
                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Eliminar()
            {
                try
                {
                    if (this.rclId != 0)
                    {
                        ocdGestor.EjecutarNonQuery("[RecuperarClave.Eliminar]", new object[,] { {
                            "@rclId",
                            rclId
                        } });
                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public int Insertar()
            {
                int IdMax = 0;
                try
                {
                    IdMax = ocdGestor.EjecutarNonQueryRetornaId("[RecuperarClave.Insertar]", new object[,] {
                        {
                            "@rclFecha",
                            rclFecha
                        },
                        {
                            "@rclUsuario",
                            rclUsuario
                        },
                        {
                            "@rclEmailRecuperacion",
                            rclEmailRecuperacion
                        },
                        {
                            "@rclUsuarioEncriptado",
                            rclUsuarioEncriptado
                        },
                        {
                            "@rclEmailRecuperacionEncriptado",
                            rclEmailRecuperacionEncriptado
                        },
                        {
                            "@rclActivo",
                            rclActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@rclFechaHoraCreacion",
                            rclFechaHoraCreacion
                        },
                        {
                            "@rclFechaHoraUltimaModificacion",
                            rclFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                return IdMax;
            }

        }
    }
}

File: Negocio/RecuperarClave.vb
Namespace LiquidacionSueldos
    Namespace Negocio
        Partial Public Class RecuperarClave

            Private ocdGestor As New LiquidacionSueldos.Datos.Gestor
            Private Fila As DataTable
            Private Tabla As New DataTable

			Private _rclId As Integer
            Public Property rclId() As Integer
                Get
					Return _rclId
                End Get
                Set(ByVal value As Integer)
                    _rclId = value
                End Set
            End Property

			Private _rclFecha As Date
            Public Property rclFecha() As Date
                Get
					Return _rclFecha
                End Get
                Set(ByVal value As Date)
                    _rclFecha = value
                End Set
            End Property

			Private _rclUsuario As String
            Public Property rclUsuario() As String
                Get
					Return _rclUsuario
                End Get
                Set(ByVal value As String)
                    _rclUsuario = value
                End Set
            End Property

			Private _rclEmailRecuperacion As String
            Public Property rclEmailRecuperacion() As String
                Get
					Return _rclEmailRecuperacion
                End Get
                Set(ByVal value As String)
                    _rclEmailRecuperacion = value
                End Set
            End Property

			Private _rclUsuarioEncriptado As String
            Public Property rclUsuarioEncriptado() As String
                Get
					Return _rclUsuarioEncriptado
                End Get
                Set(ByVal value As String)
                    _rclUsuarioEncriptado = value
                End Set
            End Property

			Private _rclEmailRecuperacionEncriptado As String
            Public Property rclEmailRecuperacionEncriptado() As String
                Get
					Return _rclEmailRecuperacionEncriptado
                End Get
                Set(ByVal value As String)
                    _rclEmailRecuperacionEncriptado = value
                End Set
            End Property

			Private _rclActivo As Boolean
            Public Property rclActivo() As Boolean
                Get
					Return _rclActivo
                End Get
                Set(ByVal value As Boolean)
                    _rclActivo = value
                End Set
            End Property

			Private _rclFechaHoraCreacion As Date
            Public Property rclFechaHoraCreacion() As Date
                Get
					Return _rclFechaHoraCreacion
                End Get
                Set(ByVal value As Date)
                    _rclFechaHoraCreacion = value
                End Set
            End Property

			Private _rclFechaHoraUltimaModificacion As Date
            Public Property rclFechaHoraUltimaModificacion() As Date
                Get
					Return _rclFechaHoraUltimaModificacion
                End Get
                Set(ByVal value As Date)
                    _rclFechaHoraUltimaModificacion = value
                End Set
            End Property

			Private _usuIdCreacion As Integer
            Public Property usuIdCreacion() As Integer
                Get
					Return _usuIdCreacion
                End Get
                Set(ByVal value As Integer)
                    _usuIdCreacion = value
                End Set
            End Property

			Private _usuIdUltimaModificacion As Integer
            Public Property usuIdUltimaModificacion() As Integer
                Get
					Return _usuIdUltimaModificacion
                End Get
                Set(ByVal value As Integer)
                    _usuIdUltimaModificacion = value
                End Set
            End Property

			Public Sub New()
                Try                
                    Me.rclId = 0
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub            

			Public Sub New(ByVal rclId As Integer)
				Try
                    Fila = New DataTable
                    Fila = ocdGestor.EjecutarReader("[RecuperarClave.ObtenerUno]", New Object(,) {{"@rclId", rclId}})

					If Fila.Rows.Count > 0 Then
						If Fila(0)("rclId").ToString().Trim().Length > 0 Then
                            Me.rclId = Convert.ToInt32(Fila(0)("rclId"))
                        Else
                            Me.rclId = 0
                        End If
                        
						If Fila(0)("rclFecha").ToString().Trim().Length > 0 Then
                            Me.rclFecha = Convert.ToDateTime(Fila(0)("rclFecha"))
                        Else
                            Me.rclFecha = Date.Now
                        End If
                        
						If Fila(0)("rclUsuario").ToString().Trim().Length > 0 Then
                            Me.rclUsuario = Convert.ToString(Fila(0)("rclUsuario"))
                        Else
                            Me.rclUsuario = ""
                        End If
                        
						If Fila(0)("rclEmailRecuperacion").ToString().Trim().Length > 0 Then
                            Me.rclEmailRecuperacion = Convert.ToString(Fila(0)("rclEmailRecuperacion"))
                        Else
                            Me.rclEmailRecuperacion = ""
                        End If
                        
						If Fila(0)("rclUsuarioEncriptado").ToString().Trim().Length > 0 Then
                            Me.rclUsuarioEncriptado = Convert.ToString(Fila(0)("rclUsuarioEncriptado"))
                        Else
                            Me.rclUsuarioEncriptado = ""
                        End If
                        
						If Fila(0)("rclEmailRecuperacionEncriptado").ToString().Trim().Length > 0 Then
                            Me.rclEmailRecuperacionEncriptado = Convert.ToString(Fila(0)("rclEmailRecuperacionEncriptado"))
                        Else
                            Me.rclEmailRecuperacionEncriptado = ""
                        End If
                        
						If Fila(0)("rclFechaHoraCreacion").ToString().Trim().Length > 0 Then
                            Me.rclFechaHoraCreacion = Convert.ToDateTime(Fila(0)("rclFechaHoraCreacion"))
                        Else
                            Me.rclFechaHoraCreacion = Date.Now
                        End If
                        
						If Fila(0)("rclFechaHoraUltimaModificacion").ToString().Trim().Length > 0 Then
                            Me.rclFechaHoraUltimaModificacion = Convert.ToDateTime(Fila(0)("rclFechaHoraUltimaModificacion"))
                        Else
                            Me.rclFechaHoraUltimaModificacion = Date.Now
                        End If
                        
						If Fila(0)("rclActivo").ToString().Trim().Length > 0 Then
                            Me.rclActivo = IIf(Convert.ToInt32(Fila(0)("rclActivo")) = 1, True, False)
                        Else
							Me.rclActivo = False
                        End If
                        
						If Fila(0)("usuIdCreacion").ToString().Trim().Length > 0 Then
                            Me.usuIdCreacion = Convert.ToInt32(Fila(0)("usuIdCreacion"))
                        Else
							Me.usuIdCreacion = 0
                        End If
                        
						If Fila(0)("usuIdUltimaModificacion").ToString().Trim().Length > 0 Then
                            Me.usuIdUltimaModificacion = Convert.ToInt32(Fila(0)("usuIdUltimaModificacion"))
                        Else
							Me.usuIdUltimaModificacion = 0
                        End If
                        
					End If
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Sub New(ByVal rclId As Integer, ByVal rclFecha As Date, ByVal rclUsuario As String, ByVal rclEmailRecuperacion As String, ByVal rclUsuarioEncriptado As String, ByVal rclEmailRecuperacionEncriptado As String, ByVal rclActivo As Boolean, ByVal rclFechaHoraCreacion As Date, ByVal rclFechaHoraUltimaModificacion As Date, ByVal IusuIdCreacion As Integer, ByVal IusuIdUltimaModificacion As Integer)
				Try
					me.rclId = rclId
					me.rclFecha = rclFecha
					me.rclUsuario = rclUsuario
					me.rclEmailRecuperacion = rclEmailRecuperacion
					me.rclUsuarioEncriptado = rclUsuarioEncriptado
					me.rclEmailRecuperacionEncriptado = rclEmailRecuperacionEncriptado
					me.rclActivo = rclActivo
					me.rclFechaHoraCreacion = rclFechaHoraCreacion
					me.rclFechaHoraUltimaModificacion = rclFechaHoraUltimaModificacion
					me.usuIdCreacion = usuIdCreacion
					me.usuIdUltimaModificacion = usuIdUltimaModificacion
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			
			Public Function ObtenerConValoresEncriptados(ByVal UsuarioEncriptado as String, ByVal EmailEncriptado as String) As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[RecuperarClave.ObtenerConValoresEncriptados]", New Object(,) {{"@UsuarioEncriptado", UsuarioEncriptado}, {"@EmailEncriptado", EmailEncriptado}})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerTodo() As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[RecuperarClave.ObtenerTodo]", New Object(,) {})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerUno(ByVal rclId as Integer) As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[RecuperarClave.ObtenerUno]", New Object(,) {{"@rclId", rclId}})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Sub Actualizar(ByVal rclId as Integer, ByVal rclFecha as Date, ByVal rclUsuario as String, ByVal rclEmailRecuperacion as String, ByVal rclUsuarioEncriptado as String, ByVal rclEmailRecuperacionEncriptado as String, ByVal rclActivo as Boolean, ByVal usuIdCreacion as Integer, ByVal usuIdUltimaModificacion as Integer, ByVal rclFechaHoraCreacion as Date, ByVal rclFechaHoraUltimaModificacion as Date)
				Try
                    ocdGestor.EjecutarNonQuery("[RecuperarClave.Actualizar]", New Object(,) {{"@rclId", rclId}, {"@rclFecha", rclFecha}, {"@rclUsuario", rclUsuario}, {"@rclEmailRecuperacion", rclEmailRecuperacion}, {"@rclUsuarioEncriptado", rclUsuarioEncriptado}, {"@rclEmailRecuperacionEncriptado", rclEmailRecuperacionEncriptado}, {"@rclActivo", rclActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@rclFechaHoraCreacion", rclFechaHoraCreacion}, {"@rclFechaHoraUltimaModificacion", rclFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Eliminar(ByVal rclId as Integer)
				Try
                    ocdGestor.EjecutarNonQuery("[RecuperarClave.Eliminar]", New Object(,) {{"@rclId", rclId}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Insertar(ByVal rclFecha as Date, ByVal rclUsuario as String, ByVal rclEmailRecuperacion as String, ByVal rclUsuarioEncriptado as String, ByVal rclEmailRecuperacionEncriptado as String, ByVal rclActivo as Boolean, ByVal usuIdCreacion as Integer, ByVal usuIdUltimaModificacion as Integer, ByVal rclFechaHoraCreacion as Date, ByVal rclFechaHoraUltimaModificacion as Date)
				Try
                    ocdGestor.EjecutarNonQuery("[RecuperarClave.Insertar]", New Object(,) {{"@rclFecha", rclFecha}, {"@rclUsuario", rclUsuario}, {"@rclEmailRecuperacion", rclEmailRecuperacion}, {"@rclUsuarioEncriptado", rclUsuarioEncriptado}, {"@rclEmailRecuperacionEncriptado", rclEmailRecuperacionEncriptado}, {"@rclActivo", rclActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@rclFechaHoraCreacion", rclFechaHoraCreacion}, {"@rclFechaHoraUltimaModificacion", rclFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub InsertarCustom(ByVal Usuario as String, ByVal Email as String, ByVal UsuarioEncriptado as String, ByVal EmailEncriptado as String)
				Try
                    ocdGestor.EjecutarNonQuery("[RecuperarClave.InsertarCustom]", New Object(,) {{"@Usuario", Usuario}, {"@Email", Email}, {"@UsuarioEncriptado", UsuarioEncriptado}, {"@EmailEncriptado", EmailEncriptado}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Actualizar()
                Try
                    If Me.rclId <> 0 Then
                        ocdGestor.EjecutarNonQuery("[RecuperarClave.Actualizar]", New Object(,) {{"@rclId", rclId}, {"@rclFecha", rclFecha}, {"@rclUsuario", rclUsuario}, {"@rclEmailRecuperacion", rclEmailRecuperacion}, {"@rclUsuarioEncriptado", rclUsuarioEncriptado}, {"@rclEmailRecuperacionEncriptado", rclEmailRecuperacionEncriptado}, {"@rclActivo", rclActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@rclFechaHoraCreacion", rclFechaHoraCreacion}, {"@rclFechaHoraUltimaModificacion", rclFechaHoraUltimaModificacion}})
                    End If
                Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Sub Eliminar()
                Try
                    If Me.rclId <> 0 Then
                        ocdGestor.EjecutarNonQuery("[RecuperarClave.Eliminar]", New Object(,) {{"@rclId", rclId}})
                    End If
                Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Function Insertar() as Integer
                Dim IdMax As Integer
                Try
                    IdMax = ocdGestor.EjecutarNonQueryRetornaId("[RecuperarClave.Insertar]", New Object(,) {{"@rclFecha", rclFecha}, {"@rclUsuario", rclUsuario}, {"@rclEmailRecuperacion", rclEmailRecuperacion}, {"@rclUsuarioEncriptado", rclUsuarioEncriptado}, {"@rclEmailRecuperacionEncriptado", rclEmailRecuperacionEncriptado}, {"@rclActivo", rclActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@rclFechaHoraCreacion", rclFechaHoraCreacion}, {"@rclFechaHoraUltimaModificacion", rclFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
                Return IdMax
            End Function

		End Class
    End Namespace
End Namespace